core/aws-core/src/main/java/software/amazon/awssdk/awscore/endpoint/AwsClientEndpointProvider.java

```diff
@@ -156,6 +166,21 @@ private Optional<URI> profileProperty(Builder builder, String profileProperty) {
                                  .flatMap(p -> p.property(profileProperty)));
     }
 
+    private Optional<URI> servicesProperty(Builder builder) {
+        Optional<ProfileFile> profileFile = Optional.ofNullable(builder.profileFile.get());
+        Optional<String> servicesSectionName = profileFile
+            .flatMap(pf -> pf.profile(builder.profileName))
+            .flatMap(p -> p.property("services"));
+
+        Optional<URI> serviceEndpoint = servicesSectionName
+            .flatMap(name -> profileFile.flatMap(pf -> pf.getSection("services", name)))
+            .flatMap(p -> Optional.ofNullable(p.properties().get(builder.serviceProfileProperty
+                                                                 + "." + ProfileProperty.ENDPOINT_URL)))
```

Comment:
You could have used a normal `map` here and not wrap it in `Optional`. 

Is there a reason you're not using `p.property(builder.serviceProfileProperty + "." + ProfileProperty.ENDPOINT_URL)`?

Reply:
No particular reason, just unfamiliar with this API. Fixed now.